spec/support/shared_examples/social_security_number.rb

```diff
@@ -0,0 +1,39 @@
+RSpec.shared_examples "social security number" do
+  describe "social security number" do
+    it "allows nils" do
+      subject.ssn = nil
+      expect(subject).to be_valid
+    end
+
+    it "allows ssns with no delimiters" do
+      subject.ssn = "123121234"
+      expect(subject).to be_valid
+    end
+
+    it "disallows bogus ssns" do
+      subject.ssn = "BOGUS"
+      expect(subject).not_to be_valid
+    end
+
+    it "invalidates bad SSNs with a friendly message" do
+      subject.ssn = "111 22 333"
+      expect(subject).not_to be_valid
+      expect(subject.errors[:ssn]).to include(
+        "Make sure your SSN has 9 digits",
+      )
+    end
+
+    it "disallows bogus ssns with newlines" do
+      subject.ssn = "BOGUS\n'--DROP TABLE CALFRESH_APPLICATIONS"
+      expect(subject).not_to be_valid
+    end
+
+    it "does not allow dashes" do
+      subject.ssn = "  123-12-1234  "
+      expect(subject).to be_invalid
+      expect(subject.errors[:ssn]).to include(
+        "Make sure your SSN has 9 digits",
+      )
+    end
+  end
+end
```

Comment:
@jayroh no, it just strips whitespace. https://github.com/holli/auto_strip_attributes

the only thing I don't love about only allowing digits (I think that is what you are referring to when you say masking) is that it can sometimes feel unexpected. For example, our financial fields only allow numbers. Yesterday I was typing some words in there and nothing happened...so I thought my external keyboard was broken. Then I realized that numbers did work.

On the other hand, I can totally see why we'd want to help people not enter invalid info.

I think whether we want to do that is a UX question?